libs/net/services/ServiceState.cs

```diff
@@ -45,7 +45,7 @@ public int RecordFailure()
     {
         this.Failures++;
 
-        if (this.Failures >= this.MaxRetryLimit) this.Status = ServiceStatus.Sleeping;
+        if (this.Failures >= this.MaxRetryLimit) this.Status = ServiceStatus.RequestSleep;
```

Comment:
Need transition status so that the service manager can ask service actions to stop.